fdbserver/DDTeamCollection.actor.cpp

```diff
@@ -159,26 +161,34 @@ class DDTeamCollectionImpl {
 	static void getTeamByServers(DDTeamCollection* self, GetTeamRequest req) {
 		const std::string servers = TCTeamInfo::serversToString(req.src);
 		Optional<Reference<IDataDistributionTeam>> res;
-		for (const auto& team : self->teams) {
-			if (team->getServerIDsStr() == servers) {
-				res = team;
-				break;
+		bool found = false;
+		for (const auto& teamSet : self->teamSets.teamSets()) {
+			for (const auto& team : teamSet->teams()) {
+				if (team->getServerIDsStr() == servers) {
+					res = team;
+					found = true;
+					break;
+				}
 			}
+			if (found)
+				break;
 		}
 		req.reply.send(std::make_pair(res, false));
 	}
 
 	// SOMEDAY: Make bestTeam better about deciding to leave a shard where it is (e.g. in PRIORITY_TEAM_HEALTHY case)
 	//		    use keys, src, dest, metrics, priority, system load, etc.. to decide...
-	ACTOR static Future<Void> getTeam(DDTeamCollection* self, GetTeamRequest req) {
+	ACTOR static Future<Void> getTeam(DDTeamCollection* self,
+	                                  std::vector<Reference<TCTeamInfo>> teams,
+	                                  GetTeamRequest req) {
 		try {
 			wait(self->checkBuildTeams());
 			// report the median available space
 			if (now() - self->lastMedianAvailableSpaceUpdate > SERVER_KNOBS->AVAILABLE_SPACE_UPDATE_DELAY) {
 				self->lastMedianAvailableSpaceUpdate = now();
 				std::vector<double> teamAvailableSpace;
-				teamAvailableSpace.reserve(self->teams.size());
-				for (const auto& team : self->teams) {
+				teamAvailableSpace.reserve(teams.size());
+				for (const auto& team : teams) {
 					if (team->isHealthy()) {
 						teamAvailableSpace.push_back(team->getMinAvailableSpaceRatio());
 					}
```

Comment:
Given that this function now just iterator over teams rather than `self->teams`, I'm not sure whether this `wait(self->checkBuildTeams())` is still necessary and whether the following update `self->medianAvailableSpace` will become inaccurate because `teams` might just be a subset of `self->teams`.
Now there's just 1 team set I guess it's fine, but we'd better leave a FIXME here.

Reply:
checkBuildTeams would still involve checking across all teamSets, I think. It would be across the teamCollection whenever we want to revisit the number of teams in the collection. I think this definition of medianSpaceAvailable is relevant because we can only make movement decisions within the team in any case.